backsplash-core/src/main/scala/com/rasterfoundry/backsplash/Implicits.scala

```diff
@@ -23,36 +25,95 @@ trait Implicits
     extends ToTmsReificationOps
     with ToExtentReificationOps
     with ToHasRasterExtentsOps
-    with ToProjectStoreOps {
+    with ToProjectStoreOps
+    with ToToolStoreOps {
+
+  implicit def paintableToolTmsReification[B: TmsReification] =
+    new TmsReification[PaintableTool[B]] {
+      def kind(self: PaintableTool[B]): MamlKind = MamlKind.Image
+
+      def tmsReification(self: PaintableTool[B], buffer: Int)(
+          implicit contextShfit: ContextShift[IO]
+      ): (Int, Int, Int) => IO[Literal] = (z: Int, x: Int, y: Int) => {
+        val extent = BacksplashImage.tmsLevels(z).mapTransform.keyToExtent(x, y)
+        val layerEval = LayerTms.apply(IO.pure(self.expr),
+                                       IO.pure(self.paramMap),
+                                       self.interpreter)
+        layerEval(z, x, y) map {
+          case Valid(tile) => RasterLit(Raster(self.painter(tile), extent))
```

Comment:
Do we want to paint prior to evaluation of the AST? I'm thinking no

Reply:
I was under the impression that `layerEval` would evaluate the AST -- is that not what's happening there?